WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -121,10 +123,19 @@ public void onMessageReceived(String from, Bundle data) {
             return;
         }
 
-        // Handle helpshift PNs
-        if (TextUtils.equals(data.getString("origin"), "helpshift")) {
-            // TODO: Handle Zendesk PNs instead
-//            HelpshiftHelper.getInstance().handlePush(this, new Intent().putExtras(data));
+        // Handle Zendesk PNs
+        // TODO: Move the contents of this to a helper
+        if (TextUtils.equals(data.getString("type"), "zendesk")) {
+            // TODO: Is it possible to keep the current `WPMainActivity`? App opening animation can be distracting
```

Comment:
Nope - they don't have a handler. As much as I hate their SDK, I think they made some correct decisions (maybe by mistake?) and I think this is one of them.

If our notification code was a little better, it'd be like a single method call to create the notification. I always hate libraries that tries to do too much, because they have to assume too much to do that and although it works out very well when the assumptions are correct, it becomes a big headache when they are not.

Reply:
> they have to assume too much to do that and although it works out very well when the assumptions are correct, it becomes a big headache when they are not.

💯